src/pybind/mgr/dashboard/frontend/src/app/shared/api/prometheus.service.ts

```diff
@@ -200,36 +206,71 @@ export class PrometheusService {
           this.timerGetPrometheusDataSub.unsubscribe();
         }
 
         this.timerGetPrometheusDataSub = timer(0, this.timerTime).subscribe(() => {
-          selectedTime = this.updateTimeStamp(selectedTime);
+          if (selectedTime) {
+            selectedTime = this.updateTimeStamp(selectedTime);
+          }
+          if (poolsChartSelectedTime) {
+            poolsChartSelectedTime = this.updateTimeStamp(poolsChartSelectedTime);
+          }
+          if (clusterUtilizationChartSelectedTime) {
+            clusterUtilizationChartSelectedTime = this.updateTimeStamp(
+              clusterUtilizationChartSelectedTime
+            );
+          }
+
+          if (selectedQueries) {
+            if (selectedQueries === 'clusterUtilization') {
+              queries = ClusterUltilizationQueries;
+            }
+            if (selectedQueries === 'poolUtilization') {
+              queries = PoolUltilizationQueries;
+            }
+            if (selectedQueries === 'allQueries') {
+              queries = allQueries;
+            }
+          }
 
           const requests = [];
           for (const queryName in queries) {
             if (queries.hasOwnProperty(queryName)) {
-              const validRangeQueries1 = [
-                'CLUSTER_CAPACITY_UTILIZATION',
-                'CLUSTER_IOPS_UTILIZATION',
-                'CLUSTER_THROUGHPUT_UTILIZATION',
-                'POOL_CAPACITY_UTILIZATION',
-                'POOL_IOPS_UTILIZATION',
-                'POOL_THROUGHPUT_UTILIZATION'
-              ];
-              if (validRangeQueries1.includes(queryName)) {
+              if (validRangeQueries.includes(queryName)) {
                 const query = queries[queryName];
-                const request = this.getMultiClusterQueryRangeData({
-                  params: encodeURIComponent(query),
-                  start: selectedTime['start'],
-                  end: selectedTime['end'],
-                  step: selectedTime['step']
-                });
-                requests.push(request);
+                if (PoolUltilizationQueryList.includes(queryName)) {
+                  const request = this.getMultiClusterQueryRangeData({
+                    params: encodeURIComponent(query),
+                    start: poolsChartSelectedTime['start'],
+                    end: poolsChartSelectedTime['end'],
+                    step: poolsChartSelectedTime['step']
+                  });
+                  requests.push(request);
+                } else if (ClusterUltilizationQueryList.includes(queryName)) {
+                  const request = this.getMultiClusterQueryRangeData({
+                    params: encodeURIComponent(query),
+                    start: clusterUtilizationChartSelectedTime['start'],
+                    end: clusterUtilizationChartSelectedTime['end'],
+                    step: clusterUtilizationChartSelectedTime['step']
+                  });
+                  requests.push(request);
+                }
```

Comment:
if at some point we introduce more charts apart from PoolUtilization or ClusterUtlization, then we need to grow this if else again. For what seems like a duplicated code, this could have been implemented bit more carefully and generalized way

Reply:
I can try to generalize it, but since this is working well, i'l not do it in this PR

Reply:
Updated the PR